### DISCUSSION TITLE
Suddenly stopped working

### DISCUSSION THREAD

**Discussion by GeraldPape** [johanmeijer/grott#556] [Q&A] [answered]
Hi, 
I use the Grott Beta branch (2.8) with HA extension in homeassistant.
The integration worked several months acceptable, but some hours ago it just stopped updating values for HA.
It seems that the proxy mode for growatt server is working, as growatt web site is updated, but HA shows no updates.
The log is useless, as no error is shown, no timestamps for whats going on, etc.

Any hints where to look for the cause?
Thanks!

**Comment by kenchy** [comment]
This happened to me in the UK at around 7:42am. At that point I stopped seeing packets in the grott journalctl log, so rightly or wrongly I assumed that the value of the server had reverted (or been changed) back to the original value. Especially (as you said) the app was still working. So I just changed the server value for my stick back to the 'correct' (local) value and it started working again.

Not sure why this happened.

**Comment by kenchy** [comment]
As far as I could tell, there was no way in server.growatt.com to see what that value is set to on the data logger. so I couldn't tell if that was actually the problem.

**Comment by mmalolepszy** [comment, marked as answer]
Had the same issue and it happened at the exactly same time: 8:42 CET (and I'm aware of another occurrence at my friends). In my case the server value was changed on data logger back to server.growatt.com. I tried to change it back to the IP of my grott server but upon data logger reset, after first data packet was sent, it got reverted back to growatt address. Apparently growatt started changing the server setting through a remote command. 

I have solved it by setting `blockcmd = True` in my `grott.ini` to block incoming commands.

**Reply by kenchy** [reply]
Oh OK, I had to change it twice, to make it stick. That might explain that.

**Reply by kenchy** [reply]
Mine got set back again, so I have also turned on 'blockcmd = True'. Thanks.

**Reply by GeraldPape** [reply]
same for me, took about 10 minutes to be reset again, now blockcmd=true.

**Reply by erik-lukac** [reply]
same for me, i fixed it by blocking it on fw.
but this setting in grott is much more elegant solution

**Comment by GeraldPape** [comment]
Thanks guys, will try that when I come home. Did they also change the password for the stick? I tried all I know, but I can't get in the interface form the server.growatt.com page.

**Reply by kenchy** [reply]
No it worked for me, password = growattdate so today it is growatt20240716

**Comment by GeraldPape** [comment]
Worked for me now, thank you. Was reluctant to use the set ip field, as I usually assume you would set the deices IP, but yeah, it's the servers IP, if you use name you need to use set domain.

On a side note, anyone has a list of the registers and their meaning/value? I only know register 4 to read/set the update interval in minutes.

**Reply by xerious** [reply]
https://community.symcon.de/uploads/short-url/jFo2xrUpTus03y9fWeYrp0Dbfng.pdf

This is the best i could find, however i have a SPF and all the values are slightly off for me.
Even the once's in this doc.

This command blocking is a savior for now.
However i had automation on battery loading, switching load and grid modes and that broke with it.

Would prefer a setting to just block the server setting.

I'd put my money on 1 of these 2 things:
1) they expect a new return type, if not there reset the ip(assume the logger's ip/address changed).
2) on my calls in nodered they want a 'assToken' with any value to keep the session alive

**Reply by xerious** [reply]
More info:
It seems to be calling 2 set parameters on the dongle.

Record blocked:  0136 -> can't make out what this is, it's a big blob of data
And 0118 -> this sets the server back to server.grotwatt.com.

Maybe a setting to block 0136 and 0118, what do you guys think?

**Reply by GeraldPape** [reply]
> And 0118 -> this sets the server back to server.grotwatt.com.
> 
> Maybe a setting to block 0136 and 0118, what do you guys think?

That would be awesome

**Comment by kenchy** [comment]
there's an issue https://github.com/johanmeijer/grott/issues/563
apparently the growatt server has changed, so you need to update grott.ini, everytime it sees a request to the old server, it updates to the new.

**Comment by johanmeijer** [comment]
There are multiple issue's opened for this problem (556, 563, 557, 565). 

The challenge is that Growatt resetting the shine-lan back from sending data via grott to sending data directly to growatt servers (by changing IP or URL in the shinelan).

So you do have to a setup of the shine-lan (shinewifi or shinelink) again. Or change the IP via the website/shinephone(?) configuration dialogue.

The only way to prevent this to enable command blocking in Grott so that Growatt can not change your configuration.
That can be done by blockcmd = True in the .ini or -e gblockcmd = "True" in docker run command or set this with docker compose.

Reading true the other issue I do understand it more. Grott uses by default a old or wrong IP address for your region.

There seems to be another way to prevent growatt to prevent reset the IP address and that is the right IP growatt ip address in the grott.ini by specifying grottip = or -e ggrottip = in docker.

Lets see if we can prevent this in the future by not using an ip but an URL for growatt in grott. I have to change grott a bit for that I think.

Blocking the change will work until growatt decides to decommission the IP address.

**Comment by DennisVredeveld** [comment]
I have had the exact same issue: yesterday around 8.30h no more inverter updates after months a working like a charm. Took me a while to realize it was the datalogger that had its URL reverted to the growatt server instead of my grott installation. So, happy to have found this thread! However, I struggle to apply the changes. I have updated my grott.ini:

```
~ more /config/grott/grott.ini 
[Growatt]
#ip = server.growatt.com
ip = 8.209.71.240
port = 5279

blockcmd = True
```

And have restarted the grott add-on:
```Grott settings:

_Generic:
	version:             	 2.8.3
	verbose:             	 True
	trace:               	 False
	config file:         	 /config/grott/grott.ini
	minrecl:             	 100
	decrypt:             	 True
	compat:              	 False
	invtype:             	 TL3
	invtypemap:          	 {}
	include_all:         	 False
	blockcmd:            	 True
	noipf:               	 False
	time:                	 auto
	sendbuf:             	 True
	timezone:            	 local
	valueoffset:         	 6
	offset:              	 6
	inverterid:          	 automatic
	mode:                	 proxy
	grottip              	 default
	grottport            	 5279
_MQTT:
	nomqtt               	 False
	mqttip:              	 172.30.33.1
	mqttport:            	 1883
	mqtttopic:           	 energy/growatt
	mqttmtopic:          	 False
	mqttmtopicname:      	 energy/meter
	mqttinverterintopic: 	 False
	mqtttretain:         	 False
	mqtttauth:           	 True
	mqttuser:            	 addons
	mqttpsw:             	 **secret**
_Growatt server:
	growattip:           	 47.91.67.66
	growattport:         	 5279
_PVOutput:
	pvoutput:            	 False
	pvdisv1:             	 False
	pvtemp:              	 False
	pvurl:               	 https://pvoutput.org/service/r2/addstatus.jsp
	pvapikey:            	 yourapikey
	pvinverters:         	 1
	pvsystemid:          	 systemid1
_Influxdb:
	influx:             	 False
	influx2:            	 False
	database:           	 grottdb
	ip:                 	 localhost
	port:               	 8086
	user:               	 grott
	password:           	 **secret**
	organization:       	 grottorg
	bucket:             	 grottdb
	token:              	 **secret**
_Extension:
	extension:          	 False
	extname:            	 grottext
	extvar:             	 {'none': 'none'}


Grott proxy mode started
	 **********************************************************************************
	 - Grott - libscrc not installed, no CRC checking only record validation on length!
	 **********************************************************************************
Hostname : 8ce4abae-grott
IP :  172.30.33.8 , port :  5279 
```

I have verified the ShinePhone app is still retrieving data from the datalogger and that the datalogger still has the grott installation IP address configured, which implies grott is forwarding them correctly to Growatt. 
However, I do not see any packets received in the grott log and no `energy/growatt` topic in my MQTT explorer.
Am I missing something here?

**Reply by mmalolepszy** [reply]
As you can see in the config you pasted, grott did not pick up your ini change and has the old IP address:
```
_Growatt server:
	growattip:           	 47.91.67.66
	growattport:         	 5279
```
What's your installation? Is it an HA add-on? If so then you should change the options through add-ons UI.

**Reply by DennisVredeveld** [reply]
yes, it is the HA add-on. Updated the growattip to 8.209.71.240 but that does not seem to change anything.
ShinePhone app still reports datalogger connectivity and showing inverter output. Datalogger still has the Grott installation IP but the add-on logging does not show any packets being received.

**Reply by kenchy** [reply]
If I were you I'd double check that the datalogger is set to use grott, rather than cloud. In some cases the moment it receives a call to the old IP it is resetting the IP on the dongle.

**Reply by egguy** [reply]
Hi @DennisVredeveld, there is a slight change initiated by HA, but grott configuration is now stored in: `/addon_configs/ef7b0eff_grott/grott/` the `/config/` has been deprecated :(

The configuration fields in the add-on also works.

**Reply by DennisVredeveld** [reply]
@kenchy Yes I have noticed but it has been on my 192.168.178.xx address for all day. And afaik, the `blockcmd = True` should also avoid it from being reset to growatt servers, right?
@egguy Thanks, I did not notice that. Have copied my config file there. But I also have set them through the UI, which also reflects in the current output when restarting:
```
[16:18:06] INFO: Preparing to start...
	 **********************************************************************************
	 - Grott - libscrc not installed, no CRC checking only record validation on length!
	 **********************************************************************************
Grott Growatt logging monitor : 2.8.3

Grott process configuration file

Grott process environmental variables

	Pulled 'ginvtype=TL3' from the environment

	Pulled 'gblockcmd=true' from the environment

	Pulled 'ggrowattip=8.209.71.240' from the environment

	Pulled 'gnomqtt=False' from the environment

	Pulled 'gmqttip=172.30.33.1' from the environment

	Pulled 'gmqttport=1883' from the environment

	Pulled 'gmqttauth=True' from the environment

	Pulled 'gmqttuser=addons' from the environment

	Pulled 'gmqttpassword=juaGh3aetiena3bahw9Wu0ookaevoon8Aakaex4ohfah8ienuyi5eix6iewaepai' from the environment

Grott override settings if set in commandline

Grott process json layout files

Grott proxy mode started
	 **********************************************************************************
	 - Grott - libscrc not installed, no CRC checking only record validation on length!
	 **********************************************************************************
Hostname : 8ce4abae-grott
IP :  172.30.33.8 , port :  5279 
```

But still no updates in the log/mqtt topic :(

**Reply by egguy** [reply]
To have a better view if packets are coming, you can add the verbose mode. Also, perhaps you need to redo the configuration of the stick

**Reply by DennisVredeveld** [reply]
@egguy I assume you mean the configuration described in https://github.com/muppet3000/homeassistant-grott/blob/main/docs/setup/datalogger.md? That is just replacing the growatt server domain by my local grott IP address right? I had to do that anyway because it was reverted due to the issues described in this topic. Or do you mean another configuration of the datalogger?
I have enabled verbose mode several times, but there is no logging whatsoever after startup:
```

Grott proxy mode started
	 **********************************************************************************
	 - Grott - libscrc not installed, no CRC checking only record validation on length!
	 **********************************************************************************
Hostname : 8ce4abae-grott
IP :  172.30.33.8 , port :  5279 
```

**Reply by egguy** [reply]
If you don't get the full output of the configuration (like here: https://github.com/johanmeijer/grott/discussions/556#discussioncomment-10074748) it could be the verbose variable is overriden.

Even if not in verbose you should get some lines like:
```
 - Growatt packet received:
		  <socket.socket fd=5, family=2, type=1, proto=0, laddr=('172.30.33.7', 5279), raddr=('192.168.1.180', 56846)>

	 - Growatt packet received:
		  <socket.socket fd=4, family=2, type=1, proto=0, laddr=('172.30.33.7', 42804), raddr=('8.209.71.240', 5279)>
	 - Growatt data decrypted V2
	 - Matched inverter serial to inverter type sph
```

**Reply by DennisVredeveld** [reply]
I have finally found the solution. **Just had to restart the datalogger by powering it off and on again.** Although all settings seemed to be exactly as before, I immediately started to see packets being received in Grott. 
so maybe good to know if other people may run into similar issues!
And @egguy many thanks for your support!

**Reply by typxxi** [reply]
> Grott



> I have finally found the solution. **Just had to restart the datalogger by powering it off and on again.** Although all settings seemed to be exactly as before, I immediately started to see packets being received in Grott

UPDATE OF MY COMMENT BELOW & SHORTCUT FOR THOSE RUNNING GROTT AS HA ADD ON which is described [here](https://github.com/muppet3000/homeassistant-grott/issues/91#issuecomment-2254581671) 

ORIGINAL COMMENT
I feel totally lost cause it has happened while we were restructuring the LAN and WLAN and the whole AC wiring of 6 Growatt inverters from the old ac distribuiton box to a new and second one. When I came back days later HA did not show any data from the growatt inverters and it took quite a while till I checked every mile stone from the installation procedure till I reached the point of setting the ip adress to my grott ha addon server.

But when I had achieved that i did not recognice instantly that it had only worked once. 

And then I started to check github and grott for any hints.
Right now I still feel completely lost.

For example if you write  "Just had to restart the datalogger by powering it off and on again." I am wondering what you really mean.
Restart the datalogger from the growatt Website UI 
or 
do does "powering it off and on again" mean to shut down the inverter from the Growatt UI
or
do you simply mean turn the inverter off and on by using the RCD / curcuit breaker.

Right now I miss a simple and straight forward recipe what to do step by step which solves the problems. I have read so many discussions and endless comments that I am tired of having still the same outcome.

Running grott on HA OS I have opened the "unusual growatt setting" and enabled "gblockcmd".
Then I went back to the growatt UI and change the sticks IP to my HA Server IP running grott addon.
Has worked so far this morning until now where I got this error message 

![image](https://github.com/user-attachments/assets/0fd7ad4d-640e-42ec-b322-d0d6d00ee7ea)

**Reply by DennisVredeveld** [reply]
@typxxi I did not mean the inverter. I meant the WiFi datalogger that is wirelessly connected to your the inverter. It receives data from your inverter and sends it to growatt. It has a small power adapter which can be disconnected.

**Reply by typxxi** [reply]
thanks, but mine has no power adapter, I simply use a wifi stick. 
But I guess you meant to pull the stick / datalogger out and plug it in to repower the stick. 

I have solved my HA Grott add on issues now by simply adding the ggrowattip in the configuration which i have described [here](https://github.com/muppet3000/homeassistant-grott/issues/91#issuecomment-2254581671) 

running for a few hours right now and hope that it will be running tomorrow too. 
Key point is to avoid the blocking cause then you can not edit the inverter settings anymore (not via browser nor shinephone app)

But by adding the IP that way every thing seems to be working fine and no need to touch the grott.ini

**Comment by johanmeijer** [comment]
You will not be able to set IP address if blockcmd = True. So this this works fine I think. 

There is an other parameter in grott noipf (gnoipf) setting this to True will allow Growatt to change ip address (and block all other commands if blockcmd = True). But that will not helps with this problem ;)

**Comment by tabascoz** [comment]
in addition to any blocking capabilities in Grott, I am implementing full block on my network and firewalls to any external IP address that the inverters may try to speak to. Including Chinese companies.
